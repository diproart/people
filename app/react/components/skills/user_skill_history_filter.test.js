import React from 'react';
import UserSkillHistoryFilter from './user_skill_history_filter';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';

test('component structure match snapshot', () => {
  const cssNamespace = 'test-subject';
  const subject = renderer.create(
    <UserSkillHistoryFilter cssNamespace={cssNamespace} />
  );

  expect(subject.toJSON()).toMatchSnapshot();
});

test('component structure without any parameter match snapshot', () => {
  const subject = renderer.create(<UserSkillHistoryFilter />);
  expect(subject.toJSON()).toMatchSnapshot();
});

test('component structure with all parameters match snapshot', () => {
  const cssNamespace = 'test-subject';
  const listPrimaryText = 'My test list';
  const startDate = '2016-12-01';
  const endDate = '2016-12-12';
  const listItems = [
    {
      name: 'skill_1',
      isActive: true
    },
    {
      name: 'skill_2',
      isActive: false
    },
    {
      name: 'skill_3',
      isActive: false
    }
  ];
  const component = (
    <UserSkillHistoryFilter
      cssNamespace={cssNamespace}
      listItems={listItems}
      listPrimaryText={listPrimaryText}
      startDate={startDate}
      endDate={endDate}
    />
  );
  const subject = shallow(component);

  listItems.forEach(({name, isActive}, index) => {
    const listItem = subject
      .find(`.${cssNamespace}__filter-category-list-item`)
      .at(index);

    if (isActive) {
      expect(
        listItem.hasClass(`${cssNamespace}__filter-category-active-item`)
      ).toBeTruthy();
    } else {
      expect(
        listItem.hasClass(`${cssNamespace}__filter-category-active-item`)
      ).not.toBeTruthy();
    }
  });

  expect(renderer.create(component).toJSON()).toMatchSnapshot();
});
