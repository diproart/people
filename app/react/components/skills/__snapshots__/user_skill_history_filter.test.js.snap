exports[`test component structure match snapshot 1`] = `
<div
  className="test-subject">
  <div
    className="test-subject__filter-category">
    <div
      className="test-subject__filter-category-primary-text" />
    <ul
      className="test-subject__filter-category-list nav nav-tabs" />
  </div>
  <div
    className="test-subject__filter-date">
    <button
      className="test-subject__filter-date-button btn btn-primary"
      onClick={[Function]}>
      last month
    </button>
    <button
      className="test-subject__filter-date-button btn btn-primary"
      onClick={[Function]}>
      last 3 months
    </button>
    <div
      className="test-subject__filter-date-label">
      From:
    </div>
    <input
      className="test-subject__filter-date-input form-control"
      onChange={[Function]}
      type="date"
      value={undefined} />
    <div
      className="test-subject__filter-date-label">
      To:
    </div>
    <input
      className="test-subject__filter-date-input form-control"
      onChange={[Function]}
      type="date"
      value={undefined} />
  </div>
</div>
`;

exports[`test component structure with all parameters match snapshot 1`] = `
<div
  className="test-subject">
  <div
    className="test-subject__filter-category">
    <div
      className="test-subject__filter-category-primary-text">
      My test list
    </div>
    <ul
      className="test-subject__filter-category-list nav nav-tabs">
      <li
        className="test-subject__filter-category-list-item test-subject__filter-category-active-item"
        onClick={[Function]}>
        skill_1
      </li>
      <li
        className="test-subject__filter-category-list-item "
        onClick={[Function]}>
        skill_2
      </li>
      <li
        className="test-subject__filter-category-list-item "
        onClick={[Function]}>
        skill_3
      </li>
    </ul>
  </div>
  <div
    className="test-subject__filter-date">
    <button
      className="test-subject__filter-date-button btn btn-primary"
      onClick={[Function]}>
      last month
    </button>
    <button
      className="test-subject__filter-date-button btn btn-primary"
      onClick={[Function]}>
      last 3 months
    </button>
    <div
      className="test-subject__filter-date-label">
      From:
    </div>
    <input
      className="test-subject__filter-date-input form-control"
      onChange={[Function]}
      type="date"
      value="2016-12-01" />
    <div
      className="test-subject__filter-date-label">
      To:
    </div>
    <input
      className="test-subject__filter-date-input form-control"
      onChange={[Function]}
      type="date"
      value="2016-12-12" />
  </div>
</div>
`;

exports[`test component structure without any parameter match snapshot 1`] = `
<div
  className={undefined}>
  <div
    className="undefined__filter-category">
    <div
      className="undefined__filter-category-primary-text" />
    <ul
      className="undefined__filter-category-list nav nav-tabs" />
  </div>
  <div
    className="undefined__filter-date">
    <button
      className="undefined__filter-date-button btn btn-primary"
      onClick={[Function]}>
      last month
    </button>
    <button
      className="undefined__filter-date-button btn btn-primary"
      onClick={[Function]}>
      last 3 months
    </button>
    <div
      className="undefined__filter-date-label">
      From:
    </div>
    <input
      className="undefined__filter-date-input form-control"
      onChange={[Function]}
      type="date"
      value={undefined} />
    <div
      className="undefined__filter-date-label">
      To:
    </div>
    <input
      className="undefined__filter-date-input form-control"
      onChange={[Function]}
      type="date"
      value={undefined} />
  </div>
</div>
`;
